File: src/Nebula/Modules/InvoiceHub/Dto/AddressHub.cs
using System.Text.Json.Serialization;

namespace Nebula.Modules.InvoiceHub.Dto;

public class AddressHub
{
    [JsonPropertyName("ubigueo")]
    public string Ubigueo { get; set; } = string.Empty;

    [JsonPropertyName("departamento")]
    public string Departamento { get; set; } = string.Empty;

    [JsonPropertyName("provincia")]
    public string Provincia { get; set; } = string.Empty;

    [JsonPropertyName("distrito")]
    public string Distrito { get; set; } = string.Empty;

    [JsonPropertyName("urbanizacion")]
    public string Urbanizacion { get; set; } = string.Empty;

    [JsonPropertyName("direccion")]
    public string Direccion { get; set; } = string.Empty;

    [JsonPropertyName("codLocal")]
    public string CodLocal { get; set; } = string.Empty;
}

File: src/xml-sunat/src/Classes/AddressHub.php
<?php

namespace App\Classes;

class AddressHub
{
    public ?string $ubigueo = null;
    public ?string $departamento = null;
    public ?string $provincia = null;
    public ?string $distrito = null;
    public ?string $urbanizacion = null;
    public ?string $direccion = null;
    public ?string $codLocal = null;

    public static function createFromJson(string $json): self
    {
        $data = json_decode($json, true);

        $addressDto = new self();
        $addressDto->ubigueo = $data['ubigueo'] ?? null;
        $addressDto->departamento = $data['departamento'] ?? null;
        $addressDto->provincia = $data['provincia'] ?? null;
        $addressDto->distrito = $data['distrito'] ?? null;
        $addressDto->urbanizacion = $data['urbanizacion'] ?? null;
        $addressDto->direccion = $data['direccion'] ?? null;
        $addressDto->codLocal = $data['codLocal'] ?? null;

        return $addressDto;
    }
}